test/functional/wallet_miniscript_decaying_multisig_descriptor_psbt.py

```diff
@@ -0,0 +1,176 @@
+        for _ in range(10):  # we check that the first 10 generated addresses are the same for all participant's multisigs
+            receive_addresses = [multisig.getnewaddress() for multisig in participants["multisigs"]]
+            all(address == receive_addresses[0] for address in receive_addresses)
+            change_addresses = [multisig.getrawchangeaddress() for multisig in participants["multisigs"]]
+            all(address == change_addresses[0] for address in change_addresses)
+
+        self.log.info("Get a mature utxo to send to the multisig...")
+        coordinator_wallet = participants["signers"][0]
+        self.generatetoaddress(self.nodes[0], 101, coordinator_wallet.getnewaddress())
+
+        deposit_amount = 6.15
+        multisig_receiving_address = participants["multisigs"][0].getnewaddress()
+        self.log.info("Send funds to the resulting multisig receiving address...")
+        coordinator_wallet.sendtoaddress(multisig_receiving_address, deposit_amount)
+        self.generate(self.nodes[0], 1)
+        for participant in participants["multisigs"]:
+            assert_approx(participant.getbalance(), deposit_amount, vspan=0.001)
+
+        self.log.info("Send a transaction from the multisig! We can always spend with all signers!")
+        to = participants["signers"][self.N - 1].getnewaddress()
+        value = 1.5
+        self.log.info("First, make a sending transaction, created using `walletcreatefundedpsbt` (anyone can initiate this)...")
+        psbt = participants["multisigs"][0].walletcreatefundedpsbt(inputs=[], outputs={to: value}, feeRate=0.00010)
+
+        psbts = []
+        self.log.info("Now at least M users check the psbt with decodepsbt and (if OK) signs it with walletprocesspsbt...")
+        for m in range(self.M):
+            signers_multisig = participants["multisigs"][m]
+            self._check_psbt(psbt["psbt"], to, value, signers_multisig)
+            signing_wallet = participants["signers"][m]
+            partially_signed_psbt = signing_wallet.walletprocesspsbt(psbt["psbt"])
+            psbts.append(partially_signed_psbt["psbt"])
+
+        self.log.info("Finally, collect the signed PSBTs with combinepsbt, finalizepsbt, then broadcast the resulting transaction...")
+        combined = coordinator_wallet.combinepsbt(psbts)
+        finalized = coordinator_wallet.finalizepsbt(combined)
+        coordinator_wallet.sendrawtransaction(finalized["hex"])
+
+        self.log.info("Check that balances are correct after the transaction has been included in a block.")
+        self.generate(self.nodes[0], 1)
+        assert_approx(participants["multisigs"][0].getbalance(), deposit_amount - value, vspan=0.001)
+        assert_equal(participants["signers"][self.N - 1].getbalance(), value)
+
+        self.log.info("Send more transactions from the multisig as required signers decay, this time with a daisy chained signing flow (one after another in series)!")
```

Comment:
Most of the code before this one is very similar to `wallet_multisig_descriptor_psbt.py` test - `participants_create_multisigs` is different though.
As pointed out in other reviews as well, I feel this test can go in the same file and some commonalities can be extracted, but I don't think this is blocking though.

Reply:
Yeah there is def some duplicated code between the two tests. But I want to focus on making this simple and easy to understand. It's trying to demonstrate/document an example wallet and could be used as a reference. Kind of like a quick start / tutorial

If I were to combine these tests it would become harder for someone to quickly and easily follow, and would lose the purpose of being a quick example. I think it's quite common to have code duplication in these types of tests/examples for that reason and hope this is fine